src/sql/workbench/contrib/notebook/browser/cellViews/code.component.ts

```diff
@@ -190,6 +191,43 @@ export class CodeComponent extends CellView implements OnInit, OnChanges {
 		return this._activeCellId;
 	}
 
+	public selectCell() {
+		// When navigating to a code-cell while Editing a text-cell, state of the code cell is changed to non-edit mode
+		// To fix this we must select cell twice
+		// Once to go into selected mode, and once to go into edit-mode
+		// We cannot do this from notebook.component as we can't differentiate between a toolbar click and a cell click
+		for (let count = 0; count < 2; count++) {
+			if (this._model.activeCell) {
+				this._model.activeCell.isEditMode = true;
+				let editorControl = this._editor.getControl() as CodeEditorWidget;
+
+				editorControl.focus();
+				this._model.updateActiveCell(this._cellModel);
+			}
+			this.detectChanges();
+		}
+	}
+
+	// When selecting a code cell after being in another code cell that was in edit mode we must unfocus the old editor
```

Comment:
Please use xmldoc comments (/** ... */) for this so intellisense actually picks them up

Reply:
Although generally looking at these comments these are more implementation details so should go inside the function instead.